computer_vision/haralick_descriptors.py

```diff
@@ -0,0 +1,433 @@
+    inverse_difference = matrix / (1 + np.abs(sub))
+    homogeneity = matrix / (1 + np.power(sub, 2))
+    entropy = -(matrix[matrix > 0] * np.log(matrix[matrix > 0]))
+
+    # Sum values for descriptors ranging from the first one to the last,
+    # as all are their respective origin matrix and not the resulting value yet.
+    return [
+        maximum_prob,
+        correlation.sum(),
+        energy.sum(),
+        contrast.sum(),
+        dissimilarity.sum(),
+        inverse_difference.sum(),
+        homogeneity.sum(),
+        entropy.sum(),
+    ]
+
+
+def get_descriptors(
+    masks: tuple[np.ndarray, np.ndarray], coordinate: list
+) -> np.ndarray:
+    """
+    Calculate all Haralick descriptors for a sequence of
+    different co-occurrence matrices, given input masks and coordinates.
+
+    Example:
+        >>> img = np.array([[[108, 201, 72], [255, 11,  127]],
+        ...                 [[56,  56,  56], [128, 255, 107]]])
+        >>> gray = grayscale(img)
+        >>> binary = binarize(gray)
+        >>> morphological = opening_filter(binary)
+        >>> get_descriptors(binary_mask(gray, morphological), [0, 1])
+        array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
+    """
+    descriptors = np.zeros((len(masks), 8))
+    for idx, mask in enumerate(masks):
+        descriptors[idx] = haralick_descriptors(matrix_concurrency(mask, coordinate))
+
+    # Concatenate each individual descriptor into
+    # one single list containing sequence of descriptors
+    return np.concatenate(descriptors, axis=None)
+
+
+def euclidean(point_1: np.ndarray, point_2: np.ndarray) -> np.float32:
+    """
+    Simple method for calculating the euclidean distance between two points,
+    with type np.ndarray.
+
+    Example:
+        >>> a = np.array([1, 0, -2])
+        >>> b = np.array([2, -1, 1])
+        >>> euclidean(a, b)
+        3.3166247903554
+    """
+    return np.sqrt(np.sum(np.square(point_1 - point_2)))
+
+
+def get_distances(descriptors: np.ndarray, base: int) -> list[Any]:
+    """
+    Calculate all Euclidean distances between a selected base descriptor
+    and all other Haralick descriptors
+    The resulting comparison is return in decreasing order,
+    showing which descriptor is the most similar to the selected base.
+
+    Args:
+        descriptors: Haralick descriptors to compare with base index
+        base: Haralick descriptor index to use as base when calculating respective
+        euclidean distance to other descriptors.
+
+    Returns:
+        Ordered distances between descriptors
+
+    Example:
+        >>> index = 1
+        >>> img = np.array([[[108, 201, 72], [255, 11,  127]],
+        ...                 [[56,  56,  56], [128, 255, 107]]])
+        >>> gray = grayscale(img)
+        >>> binary = binarize(gray)
+        >>> morphological = opening_filter(binary)
+        >>> get_distances(get_descriptors(
+        ...                 binary_mask(gray, morphological), [0, 1]),
```

Comment:
```suggestion
        ...                 binary_mask(gray, morphological), (0, 1)),
```
Change list to tuple (see previous comments)